src/unicode.cpp

```diff
@@ -557,6 +557,178 @@ static std::vector<size_t> unicode_regex_split_stl(const std::string & text, con
     return bpe_offsets;
 }
 
+// K2 system regex patterns (from tokenization_kimi.py): 
```

Comment:
Noticed whiteline warnings while porting this PR (some spaces at the end of this line)